common/domain/dlq_message_handler_test.go

```diff
@@ -359,3 +502,49 @@ func (s *dlqMessageHandlerSuite) TestMergeMessages_IgnoreErrorOnUpdateDLQAckLeve
 	s.NoError(err)
 	s.Nil(token)
 }
+
+func (s *dlqMessageHandlerSuite) TestMergeMessages_NonDomainTask() {
+	ackLevel := int64(10)
+	lastMessageID := int64(20)
+	pageSize := 100
+	pageToken := []byte{}
+
+	// Create a task that mimics a non-domain replication task by not setting DomainTaskAttributes
+	tasks := []*types.ReplicationTask{
+		{
+			TaskType:     types.ReplicationTaskTypeDomain.Ptr(), // Still set to domain but no attributes
+			SourceTaskID: 1,
+		},
+	}
+
+	s.mockReplicationQueue.EXPECT().GetDLQAckLevel(gomock.Any()).Return(ackLevel, nil).Times(1)
+	s.mockReplicationQueue.EXPECT().GetMessagesFromDLQ(gomock.Any(), ackLevel, lastMessageID, pageSize, pageToken).
+		Return(tasks, nil, nil).Times(1)
+
+	token, err := s.dlqMessageHandler.Merge(context.Background(), lastMessageID, pageSize, pageToken)
+
+	s.NotNil(err)
+	s.IsType(&types.InternalServiceError{}, err)
+	s.Equal("Encounter non domain replication task in domain replication queue.", err.Error())
+	s.Nil(token)
+}
+
+func (s *dlqMessageHandlerSuite) TestEmitDLQSizeMetricsLoop_ErrorHandling() {
+	expectedError := fmt.Errorf("error fetching DLQ size")
+	s.mockReplicationQueue.EXPECT().GetDLQSize(gomock.Any()).Return(int64(0), expectedError).AnyTimes()
+
+	// Start the metrics loop in a goroutine
+	go s.dlqMessageHandler.emitDLQSizeMetricsLoop()
+
+	// Allow some time for the goroutine to run and tick at least once
+	time.Sleep(1 * time.Nanosecond)
+
+	// Close the done channel to signal the loop to stop
+	close(s.dlqMessageHandler.done)
+
+	// Wait a bit to ensure the loop exits
+	time.Sleep(1 * time.Nanosecond)
```

Comment:
I don't think this setup ensures [the desired err handling branch](https://github.com/uber/cadence/blob/807a0e2899b4203dbf09cc0c2ae78bf2d75f10ec/common/domain/dlqMessageHandler.go#L226) to be hit. It's a tricky one and I think we can ignore that

Reply:
yea.  if we wanted to test that we'd probably need to:
- start the goroutine
- advance the clock
- wait for the mock call to occur (use a `.DoAndReturn` to monitor this, e.g. close a channel in there)
- then close the done chan

Reply:
yeah, maybe will add some time later

Reply:
I think the suggestion would be to skip the test for now then, time.Sleep tends to cause test flakiness in CD pipelines and make everyone sad